main.rb

```diff
@@ -0,0 +1,470 @@
+# rubocop:disable Metrics/ModuleLength
+module Enumerable
+  # My each method definition
+
+  def my_each
+    return to_enum :my_each unless block_given?
+
+    arr = self
+    if arr.is_a? Array
+      0.upto(arr.length - 1) do |index|
+        yield (arr[index])
+      end
+    elsif arr.is_a? Range
+      arr.step(1) do |value|
+        yield value
+      end
+    end
+    arr
+  end
+
+  # my_each_with_index method definition
+  def my_each_with_index
+    return to_enum :my_each unless block_given?
+
+    arr = self
+    0.upto(arr.length - 1) do |ii|
+      yield arr[ii], ii
+    end
+    arr
+  end
+
+  # my_select method definition
+  def my_select
+    return to_enum :my_select unless block_given?
+
+    arr = self
+    back = []
+    arr.my_each { |x| back << x if yield x }
+    back
+  end
+
+  def param_reg(par, arr_i, t_reg, t_patt, who)
+    # who is false for my_all method call
+    # who is true for my_none method call
+    return [t_reg, t_patt] unless par != ''
+
+    if par.class == Regexp && (!par.match(arr_i).nil? ^ who)
+      t_reg += 1
+    elsif par.class != Regexp && arr_i.is_a?(Module.const_get(par.to_s))
+      t_patt += 1
+    end
+    [t_reg, t_patt]
+  end
+
+  # my_all? method definition
+  def my_all?(param = '')
+    arr = self
+    any_h = { t_b_e: 0, t_e: 0, t_rxp: 0, t_p: 0, who: false }
+
+    0.upto(arr.length - 1) do |i|
+      if param == ''
+        if block_given?
+          any_h[:t_b_e] += 1 if yield arr[i]
+        elsif !arr[i].nil? && arr[i] != false
+          any_h[:t_e] += 1
+        end
+      end
+      any_h[:t_rxp], any_h[:t_p] = param_reg(param, arr[i], any_h[:t_rxp], any_h[:t_p], false)
+    end
+    any_element_true(arr.length, any_h[:t_b_e], any_h[:t_e], any_h[:t_rxp], any_h[:t_p])
+
+    # if we reach this step and all elements are thruty, we exit true.
+  end
+
+  def no_block_count(element_check, block, arr_i)
+    element_check += 1 if (!arr_i.nil? || arr_i != false) && !block
+    element_check
+  end
+
+  # my_any? method definition
+  def my_any?(param = String)
+    arr = self
+    any_h = { true_block_elements: 0, true_elements: 0, t_rxp: 0, true_pattern: 0, who: false }
+
+    0.upto(arr.length - 1) do |i|
+      if block_given?
+        any_h[:true_block_elements] += 1 if yield arr[i]
+      end
+      any_h[:true_block_elements] = no_block_count(any_h[:true_block_elements], block_given?, arr[i])
+      any_h[:t_rxp], any_h[:true_pattern] = param_reg(param, arr[i], any_h[:t_rxp], any_h[:true_pattern], false)
+    end
+    any_h[:true_block_elements] >= 1 || any_h[:t_rxp] >= 1 || any_h[:true_pattern] >= 1
+  end
```

Comment:
- [ ] should be identical and returns the same thing as ruby's [`any?`](https://ruby-doc.org/core-2.6.4/Enumerable.html#method-i-any-3F). ([Screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/ruby/advanced_building_blocks_enumerable/my_any.png) from [Odin](https://www.theodinproject.com/courses/ruby-programming/lessons/advanced-building-blocks#assignment-2))
- [ ] when no block or argument is given returns true if at least one of the collection is not false or nil
```ruby
true_array.my_any? == true_array.any? #true
```
- [ ]  when a class is passed as an argument returns true if at least one of the collection is a member of such class
```ruby
words.my_any?(Integer) == words.any?(Integer) #true
```
- [ ] when a Regex is passed as an argument returns false if none of the collection matches the Regex
```ruby
words.my_any?(/z/) == words.any?(/z/) #true
```
- [ ] when a pattern other than Regex or a Class is given returns false if none of the collection matches the pattern
```ruby
words.my_any?('cat') == words.any?('cat') #true
```

Reply:
A)  This evaluation is working for me: true_array.my_any? == true_array.any? #true
irb(main):477:0> true_array
=> [true, true, true]
irb(main):478:0> true_array.my_any?
=> true
irb(main):479:0> true_array.any?
=> true
irb(main):480:0> true_array.any? == true_array.my_any?
=> true

B) **This evaluation now works. Now fixed:** words.my_any?(Integer) == words.any?(Integer) #true
irb(main):700:0> words
=> ["alo", "mama", "hey"]
irb(main):701:0> words.my_any?(Integer) 
=> false
irb(main):702:0> words.any?(Integer)
=> false
irb(main):703:0> words.my_any?(Integer) == words.any?(Integer)
=> true
c) This evaluation works on my end: words.my_any?(/z/) == words.any?(/z/)
 words
=> ["alo", "mama", "hey"]
irb(main):705:0> words.my_any?(/z/)
=> false
irb(main):706:0> words.any?(/z/)
=> false
irb(main):707:0> words.my_any?(/z/) == words.any?(/z/)
=> true

D) this evaluation now works on my end: words.my_any?('cat') == words.any?('cat')

irb(main):708:0> words.my_any?('cat')
=> false
irb(main):709:0> words.any?('cat')
=> false
irb(main):710:0> words.my_any?('cat') == words.any?('cat')
=> true
irb(main):711:0> words
=> ["alo", "mama", "hey"]